--- xfce4-smartpm-plugin-0.3.2.orig/src/xfcesmart.c	2008-04-18 17:26:00.000000000 +0200
+++ xfce4-smartpm-plugin-0.3.2/src/xfcesmart.c	2008-07-18 09:06:48.000000000 +0200
@@ -1,4 +1,4 @@
-/*  $Id: xfcesmart.c 4615 2008-04-18 15:25:52Z afb $
+/*  $Id: xfcesmart.c 5018 2008-06-30 20:43:06Z afb $
  *
  *  xfce4-smartpm-plugin - a package manager applet for the xfce4 panel
  *  Copyright (c) 2007 Anders F Bjorklund <afb@users.sourceforge.net>
@@ -60,6 +60,7 @@
   PACKAGE_MANAGER_YUM	= 1, /* use Yum+Extender (Fedora) */
   PACKAGE_MANAGER_APT	= 2, /* use APT+Synaptic (Ubuntu) */
   PACKAGE_MANAGER_SLAPT	= 3, /* use slapt+gslapt (Vector) */
+  PACKAGE_MANAGER_PGT = 4, /* use prt-get (CRUX) */
 };
 
 #define SMART_WEBSITE "http://smartpm.org/"
@@ -79,6 +80,10 @@
 #define SLAPT_PROGRAM /* "gslapt" */    PATH_GSLAPT
 #define SLAPT_UPDATE  /* "slapt-get" */ PATH_SLAPT_GET
 
+#define PGT_WEBSITE   "http://jw.tks6.net/files/crux/prt-get_manual.html"
+#define PGT_PROGRAM   "prt-get"
+#define PGT_UPDATE    "ports"
+
 #define TIMEOUT_TIME  1000 /* milliseconds */
 
 #define PROGRESS_TIME 5*60 /* seconds */
@@ -215,6 +220,9 @@
         case PACKAGE_MANAGER_SLAPT:
           /* TODO: determine number of new packages in Slapt channels */
           break;
+        case PACKAGE_MANAGER_PGT:
+          /* TODO: determine number of new packages in PGT channels */
+          break;
       }
     }
     fclose(file);
@@ -338,6 +346,12 @@
                 "--update" "\"" ;
       terminal = TRUE;
       break;
+    case PACKAGE_MANAGER_PGT:
+      argv[0] = PATH_SUDO;
+      argv[1] = "-u";
+      argv[2] = PGT_UPDATE;
+      terminal = TRUE;
+      break;
     default:
       return 0;
   }
@@ -352,6 +366,7 @@
     if (smart->package_manager == PACKAGE_MANAGER_YUM) title = YUM_UPDATE;
     if (smart->package_manager == PACKAGE_MANAGER_APT) title = APT_UPDATE;
     if (smart->package_manager == PACKAGE_MANAGER_SLAPT) title = SLAPT_UPDATE;
+    if (smart->package_manager == PACKAGE_MANAGER_PGT) title = PGT_UPDATE;
   
     args = g_strdup_printf("%s %s %s",
                  argv[0], argv[1], argv[2]);
@@ -502,6 +517,12 @@
               packages = atoi(p);
             /* TODO: read package names */
             break;
+           case PACKAGE_MANAGER_PGT:
+            /* 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded. */
+            if (strstr(p, " upgraded,") != NULL)
+              packages = atoi(p);
+            /* TODO: read package names */
+            break;
        }
       }
       else
@@ -525,7 +546,7 @@
         continue;
       }
       if (smart->package_manager != PACKAGE_MANAGER_SMART ||
-          strstr(p, "error: ") != NULL)
+          (strstr(p, "error: ") != NULL && strstr(p, "warning: ") != NULL))
       {
         if (strstr(p, "error: ") == 0)
           p += strlen("error: ");
@@ -630,6 +651,11 @@
       argv[1] = "--simulate";
       argv[2] = (smart->dist_upgrade) ? "--dist-upgrade" : "--upgrade";
       break;
+    case PACKAGE_MANAGER_PGT:
+      argv[0] = PGT_PROGRAM;
+      argv[1] = "--test";
+      argv[2] = (smart->dist_upgrade) ? "sysup" : "sysup";
+      break;
     default:
       return 0;
   }
@@ -716,7 +742,14 @@
         program = g_strdup_printf("%s %s", PATH_GKSU, SLAPT_PROGRAM);
       else
         program = g_strdup(SLAPT_PROGRAM);
+      break;  
+    case PACKAGE_MANAGER_PGT:
+      if (smart_program_exists(PATH_GKSU))
+        program = g_strdup_printf("%s %s", PATH_GKSU, PGT_PROGRAM);
+      else
+        program = g_strdup(PGT_PROGRAM);
       break;
+
     default:
       return;
   }
@@ -989,7 +1022,8 @@
   if (button != NULL)
   gtk_widget_set_sensitive (button,
      smart->package_manager == PACKAGE_MANAGER_APT ||
-     smart->package_manager == PACKAGE_MANAGER_SLAPT);
+     smart->package_manager == PACKAGE_MANAGER_SLAPT || 
+     smart->package_manager == PACKAGE_MANAGER_PGT);
 
   if (smart->show_application) /* change the application icon too */
   smart_size_changed (smart->plugin, smart->current_icon_size, smart);
@@ -1218,6 +1252,11 @@
     case PACKAGE_MANAGER_SLAPT:
       iconname = "xfce4-" SLAPT_PROGRAM;
       break;
+    /* TODO:   
+    case PACKAGE_MANAGER_PGT:  
+      iconname = "xfce4-" PGT_PROGRAM;
+      break;
+    */  
     default:
       iconname = "xfce-unknown";
       break;
@@ -1313,6 +1352,9 @@
     case PACKAGE_MANAGER_SLAPT:
       website = SLAPT_WEBSITE;
       break;
+    case PACKAGE_MANAGER_PGT:
+      website = PGT_WEBSITE;
+      break;
     default:
       website = "http://goodies.xfce.org/";
       break;
@@ -1510,7 +1552,8 @@
   g_object_set_data (G_OBJECT (plugin), "dist-upgrade", button);
   gtk_widget_set_sensitive (button,
      smart->package_manager == PACKAGE_MANAGER_APT ||
-     smart->package_manager == PACKAGE_MANAGER_SLAPT);
+     smart->package_manager == PACKAGE_MANAGER_SLAPT ||
+     smart->package_manager == PACKAGE_MANAGER_PGT);
 
   /* center dialog on the screen */
   gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER);
